### PR TITLE
[App Service] `az logicapp create/delete/show/list`: Add new commands to support logicapp related operations

### PR DESCRIPTION
**Pull request by parth21999**
**Description**

Add `az logicapp` commands.

**Testing Guide**
`az logicapp create` to create a logic app.
`az logicapp delete` to delete a logic app.
`az logicapp show` to show a logic app.
`az logicapp list` to list logic apps.

---

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Appservice

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -51,6 +51,8 @@
 
 from .tunnel import TunnelServer
 
+from .utils import (_get_location_from_resource_group, _list_app, _rename_server_farm_props, _get_location_from_webapp)
```
Comment (calvinsID) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:54]: nit: There are already functions imported from `.utils`, combine this line with below

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+from azure.cli.testsdk import (ScenarioTest, LocalContextScenarioTest, LiveScenarioTest, ResourceGroupPreparer,
+                               StorageAccountPreparer, JMESPathCheck, live_only)
+from azure.cli.testsdk.checkers import JMESPathPatternCheck
+
+TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
+
+# pylint: disable=line-too-long
+# In the future, for any reasons the repository get removed, the source code is under "sample-repo-for-deployment-test"
+# you can use to rebuild the repository
+TEST_REPO_URL = 'https://github.com/yugangw-msft/azure-site-test.git'
```
Comment (calvinsID) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_logicapp_commands.py:24]: nit: Unused variables

File: src/azure-cli/azure/cli/command_modules/appservice/logicapp/custom.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+    functions_version = '3'
+    runtime = None
+    runtime_version = None
+
+    if consumption_plan_location or not deployment_container_image_name:
+        runtime = 'node'
+        runtime_version = '12'
+
+    if deployment_source_url and deployment_local_git:
+        raise CLIError(
```
Comment (calvinsID) [inline src/azure-cli/azure/cli/command_modules/appservice/logicapp/custom.py:58]: `CLIError` is the old way, can we change to use more specific errors from `azure.cli.core.azclierror`, ex. `MutuallyExclusiveArgumentError`?

File: src/azure-cli/azure/cli/command_modules/appservice/logicapp/custom.py [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+        is_linux = os_type and os_type.lower() == 'linux'
+
+    else:  # apps with SKU based plan
+        if is_valid_resource_id(plan):
+            parse_result = parse_resource_id(plan)
+            plan_info = client.app_service_plans.get(parse_result['resource_group'], parse_result['name'])
+        else:
+            plan_info = client.app_service_plans.get(resource_group_name, plan)
+        if not plan_info:
+            raise CLIError("The plan '{}' doesn't exist".format(plan))
```
Comment (calvinsID) [inline src/azure-cli/azure/cli/command_modules/appservice/logicapp/custom.py:100]: Same with above CLIError comment

File: src/azure-cli/azure/cli/command_modules/appservice/commands.py [hunk 5]
```diff
@@ -461,3 +463,11 @@ def load_command_table(self, _):
     with self.command_group('staticwebapp secrets', custom_command_type=staticsite_sdk) as g:
         g.custom_command('list', 'list_staticsite_secrets')
         g.custom_command('reset-api-key', 'reset_staticsite_api_key', supports_no_wait=True)
+
+    with self.command_group('logicapp') as g:
+        g.custom_command('delete', 'delete_function_app')
+
+    with self.command_group('logicapp', custom_command_type=logicapp_custom) as g:
+        g.custom_command('create', 'create_logicapp', exception_handler=ex_handler_factory())
+        g.custom_command('list', 'list_logicapp', table_transformer=transform_web_list_output)
```
Comment (calvinsID) [inline src/azure-cli/azure/cli/command_modules/appservice/commands.py:472]: Are you able to get `list` to work, when you have no logicapps in your sub? I seem to be getting error: 
image

Comment (calvinsID) [reply src/azure-cli/azure/cli/command_modules/appservice/commands.py:472]: It repro's on cloudshell for me for `functionapp list` as well

**Review by calvinsID — CHANGES_REQUESTED** [review]
Nice, looks good! 
Minor changes - Can we also add a few more tests & examples, ex. using container image, using different os, etc

File: src/azure-cli/azure/cli/command_modules/appservice/commands.py [hunk 6]
```diff
@@ -461,3 +463,11 @@ def load_command_table(self, _):
     with self.command_group('staticwebapp secrets', custom_command_type=staticsite_sdk) as g:
         g.custom_command('list', 'list_staticsite_secrets')
         g.custom_command('reset-api-key', 'reset_staticsite_api_key', supports_no_wait=True)
+
+    with self.command_group('logicapp') as g:
+        g.custom_command('delete', 'delete_function_app')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/commands.py:468]: Do we need to add confirmation for the delete command? Such as `confirmation=True`

File: src/azure-cli/azure/cli/command_modules/appservice/logicapp/_help.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        az logicapp create -g MyResourceGroup --subscription MySubscription -p MyPlan -n MyUniqueAppName -s MyStorageAccount
+"""
+
+helps['logicapp delete'] = """
+type: command
+short-summary: Delete a logic app.
+examples:
+  - name: Delete a logic app. (autogenerated)
+    text: az logicapp delete --name MyLogicapp --resource-group MyResourceGroup --subscription MySubscription
+    crafted: true
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/logicapp/_help.py:31]: Is this code automatically generated by CodeGen? If not, please delete this tag ` crafted: true`

File: src/azure-cli/azure/cli/command_modules/appservice/logicapp/custom.py [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+                    docker_registry_server_password=None, docker_registry_server_user=None,
+                    deployment_container_image_name=None, tags=None):
+    # pylint: disable=too-many-statements, too-many-branches, too-many-locals
+    functions_version = '3'
+    runtime = None
+    runtime_version = None
+
+    if consumption_plan_location or not deployment_container_image_name:
+        runtime = 'node'
+        runtime_version = '12'
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/logicapp/custom.py:55]: >
functions_version = '3'
runtime_version = '12'

Do we need to consider putting these constants in file `_constants.py `?

File: src/azure-cli/azure/cli/command_modules/appservice/commands.py (hunk 5 above)
Comment (parth21999) [reply src/azure-cli/azure/cli/command_modules/appservice/commands.py:472]: Hmm that's strange, works for me:

`
(env) PS C:\Users\paaggarwal\Desktop\azure-cli> az logicapp list

[]
`

File: src/azure-cli/azure/cli/command_modules/appservice/commands.py (hunk 6 above)
Comment (parth21999) [reply src/azure-cli/azure/cli/command_modules/appservice/commands.py:468]: Not sure, `az webapp delete` and `az functionapp delete` don't have a confirmation flag, but `az staticwebapp delete` does. 

@panchagnula @calvinsID can you please confirm what to do here?

File: src/azure-cli/azure/cli/command_modules/appservice/logicapp/_help.py (hunk 7 above)
Comment (parth21999) [reply src/azure-cli/azure/cli/command_modules/appservice/logicapp/_help.py:31]: Not sure. I didn't generate the message using CodeGen but a lot of other help texts in appservice/_help.py seem to have that tag.

@calvinsID @panchagnula can you please confirm?

Comment (calvinsID) [reply src/azure-cli/azure/cli/command_modules/appservice/logicapp/_help.py:31]: Yeah you can just remove!

File: src/azure-cli/azure/cli/command_modules/appservice/commands.py (hunk 6 above)
Comment (parth21999) [reply src/azure-cli/azure/cli/command_modules/appservice/commands.py:468]: Added `confirmation=True` in next version. Will add to webapp and funtionapp in a separate PR.

File: src/azure-cli/azure/cli/command_modules/appservice/logicapp/_help.py (hunk 7 above)
Comment (parth21999) [reply src/azure-cli/azure/cli/command_modules/appservice/logicapp/_help.py:31]: Removed tag.

**Review by zhoxing-ms — APPROVED** [review]

**Review by calvinsID — APPROVED** [review]

**Pull request merged**